src/components/backend-ai-session-launcher.ts

```diff
@@ -1033,6 +1041,54 @@ export default class BackendAiSessionLauncher extends BackendAIPage {
     this.selectedVfolders = [];
   }
 
+  /**
+   * derive session infomation from manualImageName or selector and save it in sessionInfoObj.
+   *
+   */
+  _preProcessingSessionInfo() {
+    if (this.kernel !== undefined && this.version_selector.disabled === false) {
+      this.sessionInfoObj.isSelectorTrue = true;
+    } else {
+      this.sessionInfoObj.isSelectorTrue = false;
+    }
+
+    if (this.manualImageName?.value) {
+      this.sessionInfoObj.isManualTrue = true;
+    } else {
+      this.sessionInfoObj.isManualTrue = false;
+    }
```

Comment:
This code is same with `this.sessionInfoObj.isManualTrue = this.manualImageName?.value ? true : false`. But it's okay now, so you can go in a better direction as you think.